SRM-14105: SLO downtime (#1216)
harness/developer-hub: docs/service-reliability-management/use-service-reliability-management/slo-downtime/configure-slo-downtime.md

---
title: Create
description: Create a SLO downtime.
sidebar_position: 20
---

:::info note
Currently, this feature is behind the feature flag `SRM_DOWNTIME`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
:::

## Create new SLO downtime

To configure downtime for an SLO:

> 1. In your Harness project, navigate to the **Service Reliability Management** module, expand **PROJECT SETUP**, and then select **SLO Downtime**.  

Reviewer: ```suggestion
1. In your Harness project, navigate to **Service Reliability Management**, expand **PROJECT SETUP**, and then select **SLO Downtime**.  
```

   The SLO Downtime page appears.
2. Select **+ Downtime**.  
   The Create new SLO Downtime page appears.
   
   ![Create new SLO Downtime page](./static/create-slo-downtime-page.png)

### Name, description, and category

In the **Name, description and category** section, enter the following information:

1. In **Downtime Name**, enter a name for the downtime.
2. Optionally, you can add a description in the **Description (Optional)** field. You can also attach a tag to the downtime. To attach a tag, select the pencil icon next to the **Tags (Optional)** field, type a tag name, and then press enter.
3. In the **Select downtime category** section, chose a category that suits the reason for downtime. Available options are:
   
    - **Scheduled Maintenance**
    - **Deployment**
    - **Other**

4. Select **Next** to go to the **Schedule SLO downtime window** section.

### Schedule SLO downtime window

> Specify the start and end time for the downtime. You can choose to set the downtime for a single, **ONE TIME** event, or for a **RECURRING** schedule, such as daily, weekly, or monthly. 

Reviewer: ```suggestion
Specify the start and end time for the downtime. You can choose to set the downtime for a single **ONE TIME** event, or for a **RECURRING** schedule, such as daily, weekly, or monthly. 
```

1. Select the scheduling type. You can choose one of the following options:
   - **ONE TIME**: Select this option to set the downtime for a single, one time event.
   - **RECURRING**: Select this option to set the downtime for a recurring schedule, such as daily, weekly, or monthly.

2. Based on the downtime scheduling type you select, perform the following steps:
   

#### ONE TIME

1. From the **Timezone** dropdown list, select the timezone in which you want to schedule SLO maintenance.
2. Under **Start time**, configure the start date and time for the SLO downtime.
3. Under **End time**, you can configure when the SLO downtime should end in one of the following ways:

- **End after a certain length of time**  

>   To set this, under **End time**, select the **From Start Time** option. Then, choose the unit of time (minutes, hours, days, or weeks) and the duration you want to set for the downtime to end.

Reviewer: ```suggestion
  To set the **End time**, select the **From Start Time** option. Then, choose the unit of time (minutes, hours, days, or weeks) and the duration you want to set for the downtime to end.
```

  
  Let's say you want to schedule a one-time SLO downtime for 2 hours and 30 minutes. You have already specified the start time for the downtime in the **Start time** field. In the **From Start Time** field, you would select or enter 150 and then select **minutes** from the dropdown menu next to it. The end time for the SLO downtime will be automatically calculated based on the duration you set and the start time you specified.

  ![End after certain length of time](./static/end-after-certain-time.png)

- **End on a specific date and time**  

>   To set this, under **End time** select the radio button for the **Date and Time** option. Then, select the calendar icon to choose the date and time on which the SLO downtime should end.

Reviewer: ```suggestion
  To set this:
  
  1) Under **End time**, select **Date and Time**.
  
  2) Select the calendar icon to choose the date and time on which the SLO downtime should end.
```
I think this would work better as a set of steps, but this change is optional....

>   Let's say you want to schedule a one-time SLO downtime that ends on April 15, 2023 at 6:00 PM UTC. You have already specified the start time for the downtime in the Start time field. Under **End time**, select the radio button for the **Date and Time** option. Then, select the calendar icon next to the field to open the calendar picker. In the calendar picker, navigate to April 15, 2023 and select the date. Then, select 6:00 PM from the time picker.

Reviewer: Again, I think this would work better as a set of sub-steps or sub-bullets.
